packages/ui/src/components/Visualization/Canvas/CanvasForm.test.tsx

```diff
@@ -149,20 +172,88 @@ describe('CanvasForm', () => {
       data: {
         vizNode: {
           getComponentSchema: () => visualComponentSchema,
-        } as IVisualizationNode,
+          getBaseEntity: () => new CamelRouteVisualEntity(camelRouteJson.route),
+        } as unknown as IVisualizationNode,
       },
     };
 
     render(
       <EntitiesContext.Provider value={null}>
-        <CanvasForm selectedNode={selectedNode} />
+        <VisibleFlowsProvider>
+          <CanvasForm selectedNode={selectedNode} />
+        </VisibleFlowsProvider>
       </EntitiesContext.Provider>,
     );
 
     expect(visualComponentSchema.definition.parameters).toEqual({});
   });
 
+  it('should allow consumers to update the Camel Route ID', async () => {
+    const flowId = camelRouteVisualEntity.id;
+    const newName = 'MyNewId';
+    const dispatchSpy = jest.fn();
+    const visualFlowsApi = new VisualFlowsApi(dispatchSpy);
+    const { nodes } = CanvasService.getFlowDiagram(camelRouteVisualEntity.toVizNode());
+    selectedNode = nodes[nodes.length - 1];
+
+    render(
+      <EntitiesProvider>
+        <VisibleFlowsContext.Provider value={{ visibleFlows: { [flowId]: true }, visualFlowsApi }}>
+          <CanvasForm selectedNode={selectedNode} />
+        </VisibleFlowsContext.Provider>
+      </EntitiesProvider>,
+    );
+
+    const idField = screen.getAllByLabelText('Id', {selector: 'input'})[0];
+    act(() => {
+      fireEvent.change(idField, {target: {value: newName}});
+    });
+
+    const closeSideBarButton = screen.getByTestId('close-side-bar');
+    act(() => {
+      fireEvent.click(closeSideBarButton);
+    });
+
+    expect(camelRouteVisualEntity.id).toEqual(newName);
+    expect(dispatchSpy).toHaveBeenCalledWith({ type: 'renameFlow', flowId, newName });
+  });
+
+  it('should allow consumers to update the Kamelet name', async () => {
+    const kameletVisualEntity = new KameletVisualEntity(kameletJson);
+    const flowId = kameletVisualEntity.id;
+    const newName = 'MyNewName';
+    const dispatchSpy = jest.fn();
+    const visualFlowsApi = new VisualFlowsApi(dispatchSpy);
+    const { nodes } = CanvasService.getFlowDiagram(kameletVisualEntity.toVizNode());
+    selectedNode = nodes[nodes.length - 1];
+
+    render(
+      <EntitiesProvider>
+        <VisibleFlowsContext.Provider value={{ visibleFlows: { [flowId]: true }, visualFlowsApi }}>
+          <CanvasForm selectedNode={selectedNode} />
+        </VisibleFlowsContext.Provider>
+      </EntitiesProvider>,
+    );
+
+    const NameField = screen.getByDisplayValue('user-source');
```

Comment:
@lordrip , I am using `getByDisplayValue('user-source')` here as `getAllByLabelText('Id', {selector: 'input'})` couldnt find the target input field.